library/core/src/fmt/num.rs

```diff
@@ -199,6 +200,18 @@ static DEC_DIGITS_LUT: &[u8; 200] = b"\
       6061626364656667686970717273747576777879\
       8081828384858687888990919293949596979899";
 
+fn slice_buffer_to_str(buf: &[MaybeUninit<u8>], offset: usize) -> &str {
+    // SAFETY: All buf content since offset is set.
+    let written = unsafe { buf.get_unchecked(offset..) };
+    // SAFETY: Writes use ASCII from the lookup table exclusively.
+    unsafe {
+        str::from_utf8_unchecked(slice::from_raw_parts(
+            MaybeUninit::slice_as_ptr(written),
+            written.len(),
+        ))
+    }
+}
+
```

Comment:
this function should really be unsafe; these safety comments are impossible to check (and new callers might not check them).

Reply:
Made the function unsafe.

Reply:
I think it needs some safety comments, here and at its callers? Also, while you're here:

```
let s = buf.get_unchecked(offset..).assume_init_ref();
str::from_utf8_unchecked(s)
```

I can do it in a followup.

Reply:
I added safety comments.